### PR TITLE
refactor: make it multiplatform and move API files into `network`

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues
#145 
In order to move forward with the CLI, iOS and Android

### Solutions

Create Gradle submodules:
* cli
* android
* network

Both `cli` and `android` modules depend on `network` module and can use it to do any operation.
Both `cli` and `android` now are just performing greetings using the multiplatform code.

Most of the code inside the current `src/main` is now inside `network/src/commonMain`.
The current `src/main` was completely moved into `legacy` for easier consultation during the next stages of development.

### Testing

N/A

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: buildSrc/src/main/java/Dependencies.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+object Versions {
+    const val appCompat = "1.1.0"
+    const val ktor = "1.6.4"
+    const val coroutines = "1.5.2"
+    const val compose = "1.0.5"
+    const val activityCompose = "1.3.1"
+}
+
+object Dependencies {
```
Comment (gongracr) [inline buildSrc/src/main/java/Dependencies.kt:9]: Is there a way we can centralise the dependency versions of the root project and the submodules together?

File: build.gradle.kts [hunk 2]
```diff
@@ -1,91 +1,93 @@
...
     val junitVersion = "5.7.1"
-    val kotlinVersion = "1.5.31"
     val mockkVersion = "1.12.1"
     val jakartaVersion = "2.1.6"
     val jerseyVersion = "2.32"
     val tyrusVersion = "1.13.1"
     val ktxSerializationVersion = "1.3.0"
-    val ktorVersion = "1.6.4"
     val coroutinesVersion = "1.5.2"
     val junit4Version = "4.13"
```
Comment (gongracr) [inline build.gradle.kts:46]: Maybe move these lib versions to the `Dependencies` file?

**Review by gongracr — COMMENTED** [review]
Looking good enough to be merged. However, just out of personal preference, I'd rather rename the `network` folder to something that indicates it is a module like folder, maybe `libnetwork` or `network-module` ??

File: build.gradle.kts (hunk 2 above)
Comment (vitorhugods) [reply build.gradle.kts:46]: I left it on purpose for consultation.

But it kinda sucks.

I guess moving to `legacy` makes more sense.

File: buildSrc/src/main/java/Dependencies.kt (hunk 1 above)
Comment (vitorhugods) [reply buildSrc/src/main/java/Dependencies.kt:9]: As long as it isn't inside the `buildScript {}` block in the root `build.gradle.kts`: yes.

I haven't included everything in this file *yet*, to make it easier to see what has been migrated and what has not.

File: network/src/commonMain/kotlin/com/wire/kalium/network/exceptions/KaliumException.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+}
+
+@Serializable
+class AuthException : NetworkException {
+    constructor(message: String?, code: Int) : super(_message = message, _code = code)
+    constructor(code: Int) : super(_code = code)
+
+    constructor(
+        message: String?,
```
Comment (MohamadJaara) [inline network/src/commonMain/kotlin/com/wire/kalium/network/exceptions/KaliumException.kt:49]: Auth and HttpException are not needed now (this was part of the orginal java code)

**Review by MohamadJaara — APPROVED** [review]
looks good, great job

File: network/src/commonMain/kotlin/com/wire/kalium/network/exceptions/KaliumException.kt (hunk 3 above)
Comment (vitorhugods) [reply network/src/commonMain/kotlin/com/wire/kalium/network/exceptions/KaliumException.kt:49]: `AuthException` is being used by `AuthApiImp`. It's a bit out of the scope of this PR refactoring and changing the logic inside the `network` module.

**Pull request merged**